### ISSUE TITLE
Feature Request: insert prefix field from address book

### ISSUE THREAD

**Issue by evanhorn** [RealRaven2000/SmartTemplates#267]
I am interested in obtaining the prefix field from the address book so that I can automatically properly address my email. In other words, I would like to specify Mr., Ms., Dr., etc., in the address book, then be able to have SmartTemplates generate, e.g, Mr. Smith or Ms. Smith or Dr. Smith.

Also, while I don't have an immediate need for the suffix at this point, I could envision a time when it could be useful.

I am currently adding this to an "other" field, but it would be nice to be able to use the appropriate field in the address book.

Thanks!

**Comment by RealRaven2000** [comment]
I will look into it. Ajso need to check what CardBook does with  (how it stores) this field. In the meantime, there are 2 test template with the most important address book fields below.

[Address-Book-tests.zip](https://github.com/RealRaven2000/SmartTemplates/files/12855543/Address-Book-tests.zip)

Note that using %from% is not as good as %recipient% as "from" changes perspective depending on whether you write a new mail or reply.

**Comment by RealRaven2000** [comment]
I think this is only supported in Cardbook, is this correct? I cannot see it in the normal Address Book editor in Thunderbird 115...

**Comment by RealRaven2000** [comment]
I found it in the new  Tb 115 interface, by clicking the + buttons

I made a test version that manages the standard AB and also Cardbook, using the address fields prefix and suffix. Some code to test with:

```html
Prefix / suffix: %recipient(addressbook,prefix,nickname,suffix)%<br>
```


[smartTemplate-fx-4.4pre22.zip](https://github.com/RealRaven2000/SmartTemplates/files/14361686/smartTemplate-fx-4.4pre22.zip)

----

To test the version above, download the zip file and drag the file into Thunderbird Add-ons Manager without extracting contents.

**Comment by RealRaven2000** [comment]
Implemented in 4.4 - published 28/03/2024
